runtime/src/tiered_storage/hot.rs

```diff
@@ -123,15 +131,86 @@ impl TieredAccountMeta for HotAccountMeta {
     fn supports_shared_account_block() -> bool {
         false
     }
+
+    /// Returns the epoch that this account will next owe rent by parsing
+    /// the specified account block.  None will be returned if this account
+    /// does not persist this optional field.
+    fn rent_epoch(&self, data_block: &[u8]) -> Option<Epoch> {
+        let offset = self.optional_fields_offset(data_block);
+        if self.flags.has_rent_epoch() {
+            let epoch = *ByteBlockReader::read_type::<Epoch>(data_block, offset)?;
+            return Some(epoch);
+        }
+        None
+    }
+
+    /// Returns the account hash by parsing the specified account block.  None
+    /// will be returned if this account does not persist this optional field.
+    fn account_hash<'a>(&self, data_block: &'a [u8]) -> Option<&'a Hash> {
+        let mut offset = self.optional_fields_offset(data_block);
+        if self.flags.has_rent_epoch() {
+            offset += std::mem::size_of::<Epoch>();
+        }
+        if self.flags.has_account_hash() {
+            return ByteBlockReader::read_type::<Hash>(data_block, offset);
+        }
+        None
+    }
+
+    /// Returns the write version by parsing the specified account block.  None
+    /// will be returned if this account does not persist this optional field.
+    fn write_version(&self, data_block: &[u8]) -> Option<StoredMetaWriteVersion> {
+        let mut offset = self.optional_fields_offset(data_block);
+        if self.flags.has_rent_epoch() {
+            offset += std::mem::size_of::<Epoch>();
+        }
+        if self.flags.has_account_hash() {
+            offset += std::mem::size_of::<Hash>();
+        }
```

Comment:
One way I have been thinking about is to move some of the implementations to AccountMetaOptionalFields.

It would be something like this:

```
impl AccountMetaOptionalFields {
    ...
    pub fn rent_epoch_offset(&self, _flags: &AccountMetaFlags) -> usize {
        0
    }

    pub fn account_hash_offset(&self, flags: &AccountMetaFlags) -> usize {
        let mut offset = rent_epoch_offset(flags);
        if flags.has_account_hash() {
            offset += std::mem::size_of::<Hash>();
        }
        offset
    }

    pub fn write_version_offset(&self, flags: &AccountMetaFlags) -> usize {
        let mut offset = account_hash_offset(flags);
        if flags.has_write_version() {
            offset += std::mem::size_of::<StoredMetaWriteVersion>();
        }
        offset
    }

}
```

Does this make thing slightly better?

Reply:
Great!  I will turn this PR back to draft and create a separate PR for this first.